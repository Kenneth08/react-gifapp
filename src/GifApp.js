import { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

const GifApp = ({ defaultCategories = [] }) => {
  const [categories, setCategories] = useState(defaultCategories);

  return (
    <>
      <h2>
        GifApp <span className="by">by Murcast</span>
      </h2>
      <AddCategory setCategories={setCategories} />
      <hr></hr>

      <ol>
        {categories.map((category) => (
          <GifGrid key={category} category={category} />
        ))}
      </ol>
    </>
  );
};

export default GifApp;
