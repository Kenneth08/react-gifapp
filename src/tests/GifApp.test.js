import GifApp from "../GifApp";
import { shallow } from "enzyme";

describe("Tests into GifApp.js", () => {
  test("should match with the snapshoot", () => {
    const wrapper = shallow(<GifApp />);
    expect(wrapper).toMatchSnapshot();
  });
  test("should shown an array of categories", () => {
    const categories = ["One Punch Man", "One Piece"];
    const wrapper = shallow(<GifApp defaultCategories={categories} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("GifGrid").length).toBe(categories.length);
  });
});
