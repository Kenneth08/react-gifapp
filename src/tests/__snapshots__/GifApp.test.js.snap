// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests into GifApp.js should match with the snapshoot 1`] = `
<Fragment>
  <h2>
    GifApp 
    <span
      className="by"
    >
      by Murcast
    </span>
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Tests into GifApp.js should shown an array of categories 1`] = `
<Fragment>
  <h2>
    GifApp 
    <span
      className="by"
    >
      by Murcast
    </span>
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch Man"
      key="One Punch Man"
    />
    <GifGrid
      category="One Piece"
      key="One Piece"
    />
  </ol>
</Fragment>
`;
